lib/saluki-io/src/deser/codec/dogstatsd.rs

```diff
@@ -64,7 +64,11 @@ impl Decoder for DogstatsdCodec {
                 // If we need more data, it's not an error, so we just break out.
                 nom::Err::Incomplete(_) => unreachable!("incomplete error should not be emitted"),
                 nom::Err::Error(e) | nom::Err::Failure(e) => Err(ParseError::Structural {
-                    reason: format!("encountered error '{:?}': {:?}", e.code, e.input),
+                    reason: format!(
+                        "encountered error '{:?}' while processing msg '{:?}'",
```

Comment:
Should just be able to do `{}` here: `Cow<'a, str>` has a `Display` implementation so you avoid the nested quotes (i.e. `'"...."'`).

Reply:
Also, can we write out `message` instead of just `msg`?